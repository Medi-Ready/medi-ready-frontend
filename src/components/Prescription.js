import React from "react";

const Prescription = () => {
  return (
    <>
      <h2>Prescription</h2>
    </>
  );
};

export default Prescription;
