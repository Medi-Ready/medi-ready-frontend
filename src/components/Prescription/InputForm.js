import React, { useState, useContext } from "react";
import styled from "styled-components";

import ConfirmModal from "../Shared/Modal/ConfirmModal";
import { ModalContext } from "../../contexts/ModalContext";

import Button from "../Shared/Button";
import FlexBox from "../Shared/FlexBox";
import TextArea from "../Shared/TextArea";
import TextInput from "../Shared/TextInput";

const InputForm = ({ error, setIsSubmit, setFormData }) => {
  const { handleModal } = useContext(ModalContext);

  const [duration, setDuration] = useState("");
  const [description, setDescription] = useState("");

  const handleChange = (event, setState) => {
    setState(event.target.value);
  };

  const setForm = () => {
    setFormData({ duration, description });
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    handleModal(
      <ConfirmModal
        error={error}
        setForm={setForm}
        message="처방하시겠습니까?"
        setIsSubmit={setIsSubmit}
      />,
    );
  };

  return (
    <form onSubmit={handleSubmit}>
      <FlexBox>
        <div>
          <TextArea
            name="description"
            value={description}
            placeholder="복용지도를 입력해주세요."
            onChange={(event) => handleChange(event, setDescription)}
          />
        </div>
        <InputButtonBox>
          <TextInput
            width="60px"
            name="duration"
            label="duration"
            value={duration}
            onChange={(event) => handleChange(event, setDuration)}
          />
          <span>일분</span>
          <Button type="submit">처방</Button>
        </InputButtonBox>
      </FlexBox>
    </form>
  );
};

const InputButtonBox = styled.div`
  position: absolute;
  right: 0;
  bottom: 0;
  padding-left: 10px;

  label {
    margin-left: 20px;
  }

  span {
    margin-left: 10px;
  }

  button {
    margin-left: 30px;
  }
`;

export default InputForm;
