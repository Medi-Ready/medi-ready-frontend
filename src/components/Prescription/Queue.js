import React, { useState } from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

const Queue = ({ Badge, queue, targetUser, setTargetUserInfo }) => {
  const { data } = queue;

  const handleUser = (event) => {
    event.preventDefault();

    const targetUserInfo = {
      name: event.currentTarget.dataset.name,
      picture: event.currentTarget.dataset.picture,
      waiting: false,
    };

    setTargetUserInfo(targetUserInfo);
  };

  return (
    <Wrapper>
      <h2>Waiting List</h2>

      <ul>
        {data.length && data.map((data, index) => {
          const { name, picture } = data;

          return (
            <li key={name, index}>
              <a href="#" onClick={handleUser} data-picture={picture} data-name={name}>
                <img alt={name} src={picture} />
                <b>{name}</b>
                {
                  targetUser.name !== name ? <Badge>Waiting</Badge> : <Badge color="green">Treating</Badge>
                }
              </a>
            </li>
          );
        })}
      </ul>
    </Wrapper>
  );
};

Queue.propTypes = {
  targetUser: PropTypes.string.isRequired,
  queue: PropTypes.object.isRequired,
  Badge: PropTypes.object.isRequired,
  setTargetUserInfo: PropTypes.func.isRequired,
};

const Wrapper = styled.div`
  padding-top: 10px;

  ul {
    padding-top: 5px;

    li {
      position: relative;
      border-bottom: 1px solid #EEE;

      em {
        top: 23px;
      }
    }
  }

  a {
    display: block;
    padding: 15px 0;

      img {
        display: inline-block;
        width: 40px;
        height: 40px;
        object-fit: cover;
        border-radius: 50%;
        vertical-align: middle;
      }
    }
  }

  b {
    margin-left: 20px;
    font-size: 14px;
  }

  span {
    display: inline-block;
    margin-left: 50px;
    color: #767676;
    font-size: 14px;
  }

  em {
    position: absolute;
    right: 10px;
  }
`;

export default Queue;
