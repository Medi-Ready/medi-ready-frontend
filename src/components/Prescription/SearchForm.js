import React, { useState } from "react";
import { useMutation } from "react-query";

import styled from "styled-components";
import { postMedicine } from "../../api";

import SearchBar from "../SearchBar";
import Button from "../Shared/Button";
import TextInput from "../Shared/TextInput";

const SearchForm = ({ medicineList, setMedicineList }) => {
  const [keyword, setKeyword] = useState("");
  const [searchResult, setSearchResult] = useState([]);

  const medicineMutation = useMutation(postMedicine, {
    onSuccess: (result) => {
      const { data } = result;

      if (data) {
        setKeyword("");
        setSearchResult([]);
        setMedicineList(prevItems => [...prevItems, {
          id: data.medicine_id,
          name: data.name,
        }]);
      }
    },
  });

  const handleSearch = (event) => {
    event.preventDefault();

    if (!keyword) {
      return;
    }

    medicineMutation.mutate({ name: keyword });
  };

  return (
    <Wrapper>
      <form onSubmit={handleSearch} autoComplete="off">
        <SearchBar
          keyword={keyword}
          setKeyword={setKeyword}
          searchResult={searchResult}
          setSearchResult={setSearchResult}
        />
        <StyledButton type="submit" text="Search">추가</StyledButton>
        <div>
          {medicineList.map((item, i) => (
            <TextInput
              disabled
              key={medicineList[i].id, i}
              name="medicine"
              label={`medicine${i}`}
              value={item.name.length > 20 ? item.name.slice(0, 20) + "…" : item.name}
              readOnly
            />
          ))}
        </div>
      </form>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  position: relative;

  button {
    position: absolute;
    right: 0;
  }

  div input {
    margin-top: 10px;
  }
`;

const StyledButton = styled(Button)`
  position: absolute;
  top: 0;
  right: 0;
`;

export default SearchForm;
