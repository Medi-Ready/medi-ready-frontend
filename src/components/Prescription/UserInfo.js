import React from "react";
import styled from "styled-components";

import Badge from "../../components/Shared/Badge";

const UserInfo = ({ selectedUser }) => {
  const { name, picture } = selectedUser;

  return (
    <Wrapper>
      <span>
        <img src={picture} alt={name} />
      </span>
      {name ?
        <>
          <b>{name}</b>
          <Badge color="green">Treating</Badge>
        </> : <>
          <p>대기 환자를 선택해주세요.</p>
          <Badge>Waiting</Badge>
        </>
      }
    </Wrapper>
  );
};

const Wrapper = styled.div`
  position: relative;
  padding-bottom: 20px;

  b {
    display: inline-block;
    margin-left: 12px;
    font-size: 18px;
    font-weight: 600;
  }

  span {
    display: inline-block;
    font-size: 12px;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background-color: #e9e9e9;

    img {
        display: inline-block;
        width: 40px;
        object-fit: cover;
        border-radius: 50%;
        vertical-align: middle;
      }
  }

  p {
    display: inline-block;
    margin-left: 10px;
    font-size: 14px;
    line-height: 45px;
    vertical-align: top;
  }

  em {
    position: absolute;
    top: 0;
    right: 0;
  }
`;

export default UserInfo;
