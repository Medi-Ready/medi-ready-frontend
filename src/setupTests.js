import React from "react";
import ReactDOM from "react-dom";
import "@testing-library/jest-dom";

import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { QueryClient, QueryClientProvider } from "react-query";

import theme from "../src/styles/theme";
import GlobalStyles from "../src/styles/";
import { ModalProvider } from "../src/contexts/ModalContext";

const queryClient = new QueryClient();

export const history = createMemoryHistory({
  initialEntries: ["/"],
});

export const customRender = (children) => (
  render((
    <Router history={history}>
      <ThemeProvider theme={theme}>
        <QueryClientProvider client={queryClient}>
          <GlobalStyles />
          <ModalProvider>
            {children}
          </ModalProvider>
        </QueryClientProvider>
      </ThemeProvider>
    </Router>
  ))
);

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element) => element);
});
