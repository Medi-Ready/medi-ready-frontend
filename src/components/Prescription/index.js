import { useQuery, useMutation } from "react-query";
import React, { useState, useEffect, useContext, useCallback } from "react";

import { debounce } from "lodash";
import styled from "styled-components";
import { getMedicineNames, getQueue, postMedicine, postPrescription } from "../../api";

import { ModalContext } from "../../contexts/ModalContext";

import Queue from "./Queue";
import Badge from "../Shared/Badge";
import SearchBar from "../SearchBar";
import Button from "../Shared/Button";
import FlexBox from "../Shared/FlexBox";
import Checkbox from "../Shared/Checkbox";
import TextArea from "../Shared/TextArea";
import TextInput from "../Shared/TextInput";
import ConfirmModal from "../Shared/Modal/ConfirmModal";

const Prescription = () => {
  const [message, setMessage] = useState("");
  const [keyword, setKeyword] = useState("");
  const [medicine, setMedicine] = useState("");
  const [duration, setDuration] = useState("");
  const [description, setDescription] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const [medicineList, setMedicineList] = useState([]);
  const [doseTimeList, setDoseTimeList] = useState([]);
  const [searchResult, setSearchResult] = useState([]);
  const [targetUserInfo, setTargetUserInfo] = useState({});
  const [isPrescriptionSubmit, setIsPrescriptionSubmit] = useState(false);

  const { name, picture } = targetUserInfo;
  const { handleModal } = useContext(ModalContext);

  useEffect(() => {
    setDoseTimeList(doseTimeDataList);
  }, []);

  useEffect(() => {
    setErrorMessage("");
  }, [duration, description, targetUserInfo, medicineList, doseTimeList]);

  const onChangeHanlder = (event, setState) => {
    setState(event.target.value);
  };

  const { data, isLoading } = useQuery("queue", getQueue, {
    refetchInterval: 3000,
  });

  const medicineNamesMutation = useMutation(getMedicineNames, {
    onSuccess: (result) => {
      let { data } = result;

      if (data) {
        data = data.slice(0, 5);
        setSearchResult(data);
      }
    },
  });

  const handleKeywordChange = (newKeyword) => {
    setKeyword(newKeyword);

    if (newKeyword) {
      searchMedicines(newKeyword);
    }
  };

  const searchMedicines = useCallback(
    debounce((newValue) => {
      medicineNamesMutation.mutate({ keyword: newValue });
    }, 300),
    [],
  );

  const medicineMutation = useMutation(postMedicine, {
    onSuccess: (result) => {
      const { data } = result;

      if (data) {
        setMedicine(data);
        setKeyword("");
        setSearchResult([]);
        setMedicineList(prevItems => [...prevItems, {
          id: data.medicine_id,
          name: data.name,
        }]);
      } else {
        setErrorMessage("해당 약은 존재하지 않습니다.");
      }
    },
  });

  const prescriptionMutation = useMutation(postPrescription, {
    onSuccess: ({ result }) => {
      if (result === "success") {
        setMessage("처방전 전송이 완료되었습니다.");
      }
    },
  });

  const handleSearch = (event) => {
    event.preventDefault();

    const { search } = event.target;

    medicineMutation.mutate({ name: event.target.search.value });

    event.target.search.value = "";
  };

  const handleChange = (event) => {
    const { id, checked } = event.target;

    if (id === "all") {
      let allCheckedList = doseTimeList.map((time) => {
        return { ...time, isChecked: checked };
      });

      setDoseTimeList(allCheckedList);
    } else {
      let checkedList = doseTimeList.map((time) =>
        time.id === id ? { ...time, isChecked: checked } : time,
      );

      setDoseTimeList(checkedList);
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    const doseTimes = [];

    for (const doseTime of doseTimeList) {
      if (doseTime.isChecked) {
        doseTimes.push(doseTime.id);
      }
    }

    const medicineIdList = medicineList.map(({ id }) => id);

    handleModal(<ConfirmModal setIsPrescriptionSubmit={setIsPrescriptionSubmit} />);

    const prescriptionForm = {
      doseTimes,
      duration,
      description,
      medicines: medicineIdList,
      date: new Date().toISOString(),
      patient_id: targetUserInfo.patient_id,
    };

    if (isPrescriptionSubmit) {
      prescriptionMutation.mutate(prescriptionForm);
      setDuration("");
      setDescription("");
      setErrorMessage("");
      setMedicineList([]);
      setTargetUserInfo({});
      setIsPrescriptionSubmit(false);
      setDoseTimeList(doseTimeDataList);
    }
  };

  const doseTimeDataList = [
    { id: "morning", label: "아침", name: "doseTime" },
    { id: "lunch", label: "점심", name: "doseTime" },
    { id: "dinner", label: "저녁", name: "doseTime" },
    { id: "beforeBed", label: "취침전", name: "doseTime" },
  ];

  return (
    <>
      <FlexBox>
        <h2>Prescription</h2>
      </FlexBox>

      <Wrapper>
        <FlexBox>
          <div>
            <UserInfo>
              {name && (
                <>
                  <span>
                    <img src={picture} alt={name} />
                  </span>
                  <b>{name}</b>
                  <Badge color="green">Treating</Badge>
                </>
              )}
            </UserInfo>

            <form onSubmit={handleSearch} autocomplete="off">
              <SearchBox>
                <SearchBar
                  type="text"
                  name="search"
                  label="search"
                  keyword={keyword}
                  results={searchResult}
                  setValue={handleKeywordChange}
                  placeholder="Enter Medicine Name"
                />
                <SearchBarButton type="submit" text="Search">추가</SearchBarButton>
                <div>
                  {medicineList.map((item, i) => (
                    <TextInput
                      key={item.id, i}
                      label={`medicine${i}`}
                      value={item.name.slice(0, 35) + "…"}
                      name="medicine"
                      disabled
                    />
                  ))}
                </div>
              </SearchBox>
            </form>
          </div>

          <div>
            <CheckboxList>
              <Checkbox
                id="all"
                name="all"
                label="전체"
                checked={doseTimeList.filter(time => time?.isChecked !== true).length < 1}
                onChange={handleChange}
              />
              {doseTimeList.map((time) => {
                const { id, name, label } = time;

                return (
                  <Checkbox
                    checked={time?.isChecked || false}
                    id={id}
                    name={name}
                    label={label}
                    onChange={handleChange}
                  />
                );
              })}
            </CheckboxList>
          </div>
        </FlexBox>

        <form onSubmit={handleSubmit}>
          <FlexBox>
            <div>
              <TextArea name="description" onChange={(event) => onChangeHanlder(event, setDescription)}></TextArea>
            </div>
            <InputButtonBox>
              <TextInput width="60px" label="duration" name="duration" onChange={(event) => onChangeHanlder(event, setDuration)} />
              <span>일치</span>
              <Button type="submit">처방</Button>
            </InputButtonBox>
          </FlexBox>
        </form>
        {message && <p>처방전 전송이 완료되었습니다.</p>}
        {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
      </Wrapper>

      {!isLoading && (
        <Wrapper>
          <Queue
            queue={data}
            targetUser={targetUserInfo}
            setTargetUserInfo={setTargetUserInfo} />
        </Wrapper>
      )}
    </>
  );
};

const Wrapper = styled.div`
  margin-top: 20px;
  padding: 20px 20px 35px;
  border-radius: 10px;
  background-color: ${({ theme }) => theme.color.white};
  box-shadow: rgb(149 157 165 / 20%) 0px 8px 24px;
`;

const CheckboxList = styled.ul`
  display: inline-flex;
  flex-wrap: wrap;
  justify-content: space-between;
  max-width: 450px;
  padding-left: 15px;
  vertical-align: top;
`;

const SearchBarButton = styled(Button)`
  position: absolute;
  top: 0;
  right: 0;
`;

const UserInfo = styled.div`
  position: relative;
  padding-bottom: 20px;

  b {
    display: inline-block;
    margin-left: 12px;
    font-size: 18px;
    font-weight: 600;
  }

  span {
    display: inline-block;
    font-size: 12px;

    img {
        display: inline-block;
        width: 40px;
        object-fit: cover;
        border-radius: 50%;
        vertical-align: middle;
      }
  }

  em {
    position: absolute;
    top: 0;
    right: 0;
  }
`;

const SearchBox = styled.div`
  position: relative;

  button {
    position: absolute;
    right: 0;
  }

  div input {
    margin-top: 10px;
  }
`;

const InputButtonBox = styled.div`
  position: absolute;
  right: 0;
  bottom: 0;
  padding-left: 10px;

  > * {
    margin-left: 20px;
  }
`;

export default Prescription;
