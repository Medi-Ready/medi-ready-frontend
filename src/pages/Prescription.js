import { useMutation } from "react-query";
import React, { useState, useEffect, useContext } from "react";
import styled from "styled-components";

import { postPrescription } from "../api";
import { ModalContext } from "../contexts/ModalContext";

import Queue from "../components/Prescription/Queue";

import Loading from "../components/Shared/Loading";
import FlexBox from "../components/Shared/FlexBox";
import Checkbox from "../components/Shared/Checkbox";
import UserInfo from "../components/Prescription/UserInfo";
import { PageTitle, PageContent } from "../components/Base";
import ErrorMessage from "../components/Shared/ErrorMessage";
import InputForm from "../components/Prescription/InputForm";
import SearchForm from "../components/Prescription/SearchForm";
import ConfirmMessage from "../components/Shared/Modal/ConfirmMessage";

const Prescription = () => {
  const { handleModal } = useContext(ModalContext);

  const [error, setError] = useState("");
  const [formData, setFormData] = useState({});
  const [isSubmit, setIsSubmit] = useState(false);
  const [medicineList, setMedicineList] = useState([]);
  const [selectedUser, setSelectedUser] = useState({});
  const [doseTimeList, setDoseTimeList] = useState([]);

  useEffect(() => {
    setDoseTimeList(doseTimeDataList);
  }, []);

  useEffect(() => {
    if (isSubmit) {
      handleSubmit();
    }
  }, [isSubmit]);

  const { mutate, isLoading } = useMutation(postPrescription, {
    onSuccess({ result }) {
      if (result === "success") {
        handleModal(<ConfirmMessage text="처방전 전송이 완료되었습니다." />);
      }
    },
  });

  if (isLoading) {
    return <Loading />;
  }

  const handleChange = (event) => {
    const { id, checked } = event.target;

    if (id === "all") {
      let allCheckedList = doseTimeList.map((time) => {
        return { ...time, isChecked: checked };
      });

      setDoseTimeList(allCheckedList);
    } else {
      let checkedList = doseTimeList.map((time) =>
        time.id === id ? { ...time, isChecked: checked } : time,
      );

      setDoseTimeList(checkedList);
    }
  };

  const handleSubmit = () => {
    const doseTimes = {};

    doseTimeList.forEach(({ isChecked, id }) => {
      if (isChecked) {
        if (id === "beforeBed") {
          return doseTimes["before_bed"] = true;
        }

        doseTimes[id] = true;
      }
    });

    const prescriptionForm = {
      doseTimes,
      medicines: medicineList,
      duration: formData.duration,
      description: formData.description,
      date: new Date().toISOString(),
      patient_id: selectedUser.patient_id,
    };

    if (!selectedUser.name) {
      setError("대기 환자를 선택해주세요.");
      setIsSubmit(false);
      return;
    }

    if (!medicineList.length) {
      setError("처방할 약을 추가해주세요.");
      setIsSubmit(false);
      return;
    }

    if (!Object.keys(doseTimes).length) {
      setError("복용 시간을 체크해주세요.");
      setIsSubmit(false);
      return;
    }

    if (!formData.description) {
      setError("복약지도는 필수 항목입니다.");
      setIsSubmit(false);
      return;
    }

    if (!formData.duration) {
      setError("복용기간을 입력해주세요.");
      setIsSubmit(false);
      return;
    }

    mutate(prescriptionForm);

    setError("");
    setFormData({});
    setIsSubmit(false);
    setSelectedUser({});
    setMedicineList([]);
    setDoseTimeList(doseTimeDataList);
  };

  const doseTimeDataList = [
    { id: "morning", label: "아침", name: "doseTime" },
    { id: "lunch", label: "점심", name: "doseTime" },
    { id: "dinner", label: "저녁", name: "doseTime" },
    { id: "beforeBed", label: "취침전", name: "doseTime" },
  ];

  return (
    <PageContent>
      <PageTitle>Prescription</PageTitle>

      <BoxWrapper>
        <FlexBox>
          <div>
            <UserInfo selectedUser={selectedUser} isSubmit={isSubmit} />
            <SearchForm
              setError={setError}
              medicineList={medicineList}
              setMedicineList={setMedicineList}
            />
          </div>

          <div>
            <CheckboxList>
              <Checkbox
                id="all"
                name="all"
                label="전체"
                checked={doseTimeList.filter(time => time?.isChecked !== true).length < 1}
                onChange={handleChange}
              />
              {doseTimeList.map((time) => {
                const { id, name, label } = time;

                return (
                  <Checkbox
                    key={id}
                    checked={time?.isChecked || false}
                    id={id}
                    name={name}
                    label={label}
                    onChange={handleChange}
                  />
                );
              })}
            </CheckboxList>
          </div>
        </FlexBox>

        <InputForm
          isSubmit={isSubmit}
          setIsSubmit={setIsSubmit}
          setFormData={setFormData}
        />

        {error && <ErrorMessage>{error}</ErrorMessage>}
      </BoxWrapper>

      <BoxWrapper>
        <Queue
          isSubmit={isSubmit}
          selectedUser={selectedUser}
          setSelectedUser={setSelectedUser}
        />
      </BoxWrapper>
    </PageContent>
  );
};

const BoxWrapper = styled.div`
  margin-top: 20px;
  padding: 20px 20px 35px;
  border-radius: 10px;
  background-color: ${({ theme }) => theme.color.white};
  box-shadow: rgb(149 157 165 / 20%) 0px 8px 24px;
`;

const CheckboxList = styled.ul`
  display: inline-flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  max-width: 430px;
  padding: 0 20px;
  vertical-align: top;
  box-sizing: border-box;

  li:first-child {
    border-right: 1px solid #eee;
  }
`;

export default Prescription;
