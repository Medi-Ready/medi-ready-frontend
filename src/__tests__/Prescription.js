import React from "react";
import { screen } from "@testing-library/react";

import Prescription from "../pages/Prescription";
import { customRender } from "../setupTests";

describe("<Prescription />", () => {
  it("matches snapshot", () => {
    const { container } = customRender(<Prescription />);

    expect(container).toMatchSnapshot();
  });

  it("should render searchForm", () => {
    customRender(<Prescription />);

    expect(screen.getByPlaceholderText("약 검색어를 입력해주세요.")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("복용지도를 입력해주세요.")).toBeInTheDocument();
  });
});
