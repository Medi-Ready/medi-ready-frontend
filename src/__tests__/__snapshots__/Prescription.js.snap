// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Prescription /> matches snapshot 1`] = `
<div>
  <article
    class="sc-jSgvzq cDUafv"
  >
    <h2
      class="sc-eCstlR jQcIqZ"
    >
      Prescription
    </h2>
    <div
      class="sc-lmoMya irSTyl"
    >
      <div
        class="sc-iqHYmW jVgdcV"
      >
        <div>
          <div
            class="sc-dQoVA fBNbtu"
          >
            <span>
              <img />
            </span>
            <p>
              대기 환자를 선택해주세요.
            </p>
            <em
              class="sc-hKgJUU bkEHFh"
            >
              Waiting
            </em>
          </div>
          <div
            class="sc-iwyWTf jVcAfQ"
          >
            <form
              autocomplete="off"
            >
              <label
                for="search"
              >
                <input
                  class="sc-idOiZg clJHyJ"
                  id="search"
                  name="search"
                  placeholder="약 검색어를 입력해주세요."
                  type="text"
                  value=""
                  width="100%"
                />
              </label>
              <button
                class="sc-kstqJO jwyYIr sc-cxFLGX gYDfCX"
                type="submit"
              >
                추가
              </button>
              <div />
            </form>
          </div>
        </div>
        <div>
          <ul
            class="sc-iJuVqt ilKhaW"
          >
            <li
              class="sc-crrszt cLeXnh"
            >
              <label
                for="all"
              >
                전체
              </label>
              <input
                checked=""
                id="all"
                name="all"
                type="checkbox"
              />
            </li>
            <li
              class="sc-crrszt cLeXnh"
            >
              <label
                for="morning"
              >
                아침
              </label>
              <input
                id="morning"
                name="doseTime"
                type="checkbox"
              />
            </li>
            <li
              class="sc-crrszt cLeXnh"
            >
              <label
                for="lunch"
              >
                점심
              </label>
              <input
                id="lunch"
                name="doseTime"
                type="checkbox"
              />
            </li>
            <li
              class="sc-crrszt cLeXnh"
            >
              <label
                for="dinner"
              >
                저녁
              </label>
              <input
                id="dinner"
                name="doseTime"
                type="checkbox"
              />
            </li>
            <li
              class="sc-crrszt cLeXnh"
            >
              <label
                for="beforeBed"
              >
                취침전
              </label>
              <input
                id="beforeBed"
                name="doseTime"
                type="checkbox"
              />
            </li>
          </ul>
        </div>
      </div>
      <form>
        <div
          class="sc-iqHYmW jVgdcV"
        >
          <div>
            <textarea
              class="sc-bkzYnD bfRRxk"
              name="description"
              placeholder="복용지도를 입력해주세요."
            />
          </div>
          <div
            class="sc-dIUeWJ cQSwdf"
          >
            <label
              for="duration"
            >
              <input
                class="sc-idOiZg hUGsaj"
                id="duration"
                name="duration"
                type="text"
                value=""
                width="60px"
              />
            </label>
            <span>
              일분
            </span>
            <button
              class="sc-kstqJO jwyYIr"
              type="submit"
            >
              처방
            </button>
          </div>
        </div>
      </form>
      
    </div>
    <div
      class="sc-lmoMya irSTyl"
    >
      <div
        class="sc-fubCzh iUTfIW"
      >
        <div
          class="sc-pGacB esCgru"
        >
          <span />
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </article>
</div>
`;
